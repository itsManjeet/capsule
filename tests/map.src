println({});
let x = {integer: 103, decimal: 10.34, string: "string", boolean: true, function: fun(a, b) { let c = 23; return a + b + c + 10; }};
println(x);
println(x.integer);
println(x.decimal);
println(x.string);
println(x.boolean);
println(x.function);
println(x.function(1, 49));

x.decimal = 12.304;
println (x);