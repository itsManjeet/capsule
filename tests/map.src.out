{}
{boolean:true, decimal:10.340000, function:0000 CONST | 0002 STORE 2 'LOCAL' | 0005 POP | 0006 LOAD 0 'LOCAL' | 0009 LOAD 1 'LOCAL' | 0012 ADD | 0013 LOAD 2 'LOCAL' | 0016 ADD | 0017 CONST | 0019 ADD | 0020 RET | , integer:103, string:string}
103
10.340000
string
true
0000 CONST | 0002 STORE 2 'LOCAL' | 0005 POP | 0006 LOAD 0 'LOCAL' | 0009 LOAD 1 'LOCAL' | 0012 ADD | 0013 LOAD 2 'LOCAL' | 0016 ADD | 0017 CONST | 0019 ADD | 0020 RET | 
83
{boolean:true, decimal:12.304000, function:0000 CONST | 0002 STORE 2 'LOCAL' | 0005 POP | 0006 LOAD 0 'LOCAL' | 0009 LOAD 1 'LOCAL' | 0012 ADD | 0013 LOAD 2 'LOCAL' | 0016 ADD | 0017 CONST | 0019 ADD | 0020 RET | , integer:103, string:string}
